### ISSUE TITLE
Allow to use AAD token instead of PAT

### ISSUE THREAD

**Issue by alexott** [databrickslabs/overwatch#130]
It would be useful to add support for AAD Token on Azure instead of using PAT-only.  This will allow to use service principals for execution of jobs, etc.

**Comment by GeekSheikh** [comment]
I've thought about this before and am interested in adding support but is this supported by Databricks? The workspace needs to allow auths with AAD via the APIs, is that possible today?

**Comment by alexott** [comment]
Yes, it [works perfectly](https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/aad/), especially useful for service principals. For [example](https://github.com/alexott/databricks-nutter-projects-demo#instructions-for-repository-inside-azure-devops), I'm using it from the Azure DevOps to trigger tests.

**Comment by GeekSheikh** [comment]
awesome, feel free to cut a branch that makes the changes or we can have a session to discuss the relevant differences. I'm assuming it just has to do with the API call structure, but we can chat.

**Comment by alexott** [comment]
It’s just a different token, otherwise it’s the same

**Comment by GeekSheikh** [comment]
Then what do we have to do to "add support"?

**Comment by alexott** [comment]
We need to add following:
* allow to specify service principal & its secret
* add a code for generation of AAD token for given SPN
